raft/src/main/java/org/apache/kafka/raft/QuorumState.java

```diff
@@ -202,20 +203,40 @@ public void initialize(OffsetAndEpoch logEndOffsetAndEpoch) throws IllegalStateE
             );
         } else if (election.hasLeader()) {
             VoterSet voters = latestVoterSet.get();
-            initialState = new FollowerState(
-                time,
-                election.epoch(),
-                election.leaderId(),
-                voters.listeners(election.leaderId()),
-                voters.voterIds(),
-                Optional.empty(),
-                fetchTimeoutMs,
-                logContext
-            );
+            Endpoints leaderEndpoints = voters.listeners(election.leaderId());
+            if (leaderEndpoints.isEmpty()) {
+                // Since the leader's endpoints are not known, it cannot send Fetch or
+                // FetchSnapshot requests to the leader.
+                //
+                // Transition to unattached instead and discover the leader's endpoint through
+                // Fetch requests to the bootstrap servers or from a BeginQuorumEpoch request from
+                // the leader.
```

Comment:
do we need a log message here?

Reply:
Yes. I'll log it at "info" since it can only happen once and at startup.